### PR TITLE
bitnami/kafka add the ability to reference an existing cluster id secret

### PR DESCRIPTION
**Pull request by thomasbellio**
### Description of the change

This change allows the user of the chart to deploy Kafka in Kraft mode such that it will reference an existing Cluster ID secret in K8s. 

### Benefits

This will enable users who are deploying secret configurations through Vault or some other means so that they don't have to include potentially sensitive information like the cluster ID in source code. 

### Possible drawbacks

There don't appear to be many drawbacks to this approach, it makes the chart more flexible. One drawback might be that this adds some new complexity as do all new configuration options.  The added parameter `existingClusterIdSecret` is incompatible with explicitly setting your clusterId in the values.yaml so that might cause confusion.

### Applicable issues

N/A

### Additional information

N/A

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kafka/Chart.yaml [hunk 1]
```diff
@@ -42,4 +42,4 @@ maintainers:
 name: kafka
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/kafka
-version: 25.1.12
+version: 25.1.13
```
Comment (aoterolorenzo) [inline bitnami/kafka/Chart.yaml:45]: As this is a new feature, we should bump a minor to correctly follow [semver](https://semver.org/)

```suggestion
version: 25.2.0
```

File: bitnami/kafka/templates/broker/statefulset.yaml [hunk 2]
```diff
@@ -152,7 +152,7 @@ spec:
             - name: KAFKA_KRAFT_CLUSTER_ID
               valueFrom:
                 secretKeyRef:
-                  name: {{ printf "%s-kraft-cluster-id" (include "common.names.fullname" .) }}
+                  name: {{ ternary (printf "%s-kraft-cluster-id" (include "common.names.fullname" .)) .Values.kraft.existingClusterIdSecret  (not .Values.kraft.existingClusterIdSecret) }}
```
Comment (aoterolorenzo) [inline bitnami/kafka/templates/broker/statefulset.yaml:155]: Searching over a negated value in a ternary seems a little tricky. What about...

```suggestion
                  name: {{ default (printf "%s-kraft-cluster-id" (include "common.names.fullname" .)) .Values.kraft.existingClusterIdSecret }}
```

File: bitnami/kafka/templates/controller-eligible/statefulset.yaml [hunk 3]
```diff
@@ -151,7 +151,7 @@ spec:
             - name: KAFKA_KRAFT_CLUSTER_ID
               valueFrom:
                 secretKeyRef:
-                  name: {{ printf "%s-kraft-cluster-id" (include "common.names.fullname" .) }}
+                  name: {{ ternary (printf "%s-kraft-cluster-id" (include "common.names.fullname" .)) .Values.kraft.existingClusterIdSecret  (not .Values.kraft.existingClusterIdSecret) }}
```
Comment (aoterolorenzo) [inline bitnami/kafka/templates/controller-eligible/statefulset.yaml:154]:
```suggestion
                  name: {{ default (printf "%s-kraft-cluster-id" (include "common.names.fullname" .)) .Values.kraft.existingClusterIdSecret }}
```

**Review by aoterolorenzo — CHANGES_REQUESTED** [review]
@thomasbellio thank you for contributing! I just wrote some comments above.

**Comment by thomasbellio** [comment]
@aoterolorenzo Thanks for the suggestions, I integrated those.

**Review by aoterolorenzo — APPROVED** [review]
LGTM! Thank you @thomasbellio

**Pull request merged**